graphql/src/test/java/com/linecorp/armeria/server/graphql/GraphqlServiceMultipartTest.java

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.graphql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.google.common.io.Files;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.ContentDisposition;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.graphql.scalar.MoreScalars;
+import com.linecorp.armeria.common.multipart.BodyPart;
+import com.linecorp.armeria.common.multipart.Multipart;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import graphql.com.google.common.collect.ImmutableList;
+import graphql.schema.DataFetcher;
+import graphql.schema.idl.errors.SchemaProblem;
+
+class GraphqlServiceMultipartTest {
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final File graphqlSchemaFile =
+                    new File(getClass().getResource("/multipart.graphqls").toURI());
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      c.scalar(MoreScalars.path());
+                                      final DataFetcher<String> fileUpload = fileUploadFetcher();
+                                      c.type("Mutation",
+                                             typeWiring -> typeWiring.dataFetcher("fileUpload", fileUpload));
+                                      final DataFetcher<List<String>> fileUploads = fileUploadsFetcher();
+                                      c.type("Mutation",
+                                             typeWiring -> typeWiring.dataFetcher("fileUploads", fileUploads));
+                                  })
+                                  .build();
+            sb.service("/graphql", service);
+        }
+    };
+
+    private static DataFetcher<String> fileUploadFetcher() {
+        return environment -> {
+            final Path path = environment.getArgument("path");
+            return Files.asCharSource(path.toFile(), StandardCharsets.UTF_8).read();
```

Comment:
This would be a blocking operation. Should we execute this in a blocking task executor?

Reply:
I didn't think of this. 😅 Should I validate that users also use blocking I/O? Or is it left to the user to choose?

Then we only need to modify that test.

Reply:
We can't enforce users to use a blocking task executor. But at least we should use a correct executor because users could reference our test code as an example.

Reply:
I fixed this. Could you confirm that what you intended is correct? 🤔